.github/workflows/update_shfmt_checksums.py

```
@@ -36,37 +39,44 @@
-response = requests.get(checksum_url)
+api_url = GITHUB_API_URL.format(version=shfmt_version)
+print(f"[INFO] Fetching release metadata from: {api_url}")
+
+# Step 2: Fetch release metadata from GitHub
+# Use GITHUB_TOKEN if available to avoid the 60 req/hr unauthenticated limit;
+# GitHub Actions populates this automatically when the step has the env wired.
+headers = {}
+token = os.environ.get('GITHUB_TOKEN')
+if token:
+    headers['Authorization'] = f'Bearer {token}'
+response = requests.get(api_url, headers=headers)
```

Ruff (0.15.12)

[error] Probable use of `requests` call without timeout

(S113)